frontend/tests/controllers/CourseAssignmentsTest.php

```diff
@@ -78,4 +88,87 @@ public function testOrderAssignments() {
             );
         }
     }
+
+    public function testAllAdminsCanSeeAdminMode() {
+        $courseData = \OmegaUp\Test\Factories\Course::createCourseWithOneAssignment();
+
+        // Login admin and getting assignments list
+        $adminLogin = self::login($courseData['admin']);
+        [
+            'assignments' => $assignments,
+        ] = \OmegaUp\Controllers\Course::apiListAssignments(
+            new \OmegaUp\Request([
+                'auth_token' => $adminLogin->auth_token,
+                'course_alias' => $courseData['course_alias']
+            ])
+        );
+
+        $details = \OmegaUp\Controllers\Course::getCourseAdminDetailsForSmarty(
+            new \OmegaUp\Request([
+                'auth_token' => $adminLogin->auth_token,
+                'course_alias' => $courseData['course_alias'],
+                'assignment_alias' => $assignments[0]['alias'],
+            ])
+        );
+
+        $this->assertArrayHasKey('smartyProperties', $details);
+        $this->assertArrayHasKey('payload', $details['smartyProperties']);
+        $this->assertArrayHasKey(
+            'details',
+            $details['smartyProperties']['payload']
+        );
+
+        // A new student is added to course
+        ['identity' => $identity] = \OmegaUp\Test\Factories\User::createUser();
+
+        \OmegaUp\Test\Factories\Course::addStudentToCourse(
+            $courseData,
+            $identity
+        );
+
+        $userLogin = self::login($identity);
+
+        // Student tries to access into the course in admin mode
+        try {
+            \OmegaUp\Controllers\Course::getCourseAdminDetailsForSmarty(
+                new \OmegaUp\Request([
+                    'auth_token' => $userLogin->auth_token,
+                    'course_alias' => $courseData['course_alias'],
+                    'assignment_alias' => $assignments[0]['alias'],
+                ])
+            );
+            $this->fails('User should not have access to admin mode');
```

Comment:
nit:

```suggestion
            $this->fail('User should not have access to admin mode');
```

(aunque esto también causaría una excepción si se llega a ejecutar, fallando correctamente la prueba, pero es mejor usar el nombre correcto).